ksql-engine/src/main/java/io/confluent/ksql/engine/AuthorizationTopicAccessValidator.java

```diff
@@ -126,11 +126,8 @@ private void checkAccess(
     if (authorizedOperations != null && !authorizedOperations.contains(operation)) {
       // This error message is similar to what Kafka throws when it cannot access the topic
       // due to an authorization error. I used this message to keep a consistent message.
-      throw new KsqlException(String.format(
-              "Failed to %s Kafka topic: [%s]%n"
-                  + "Caused by: Not authorized to access topic: [%s]",
-              StringUtils.capitalize(operation.toString().toLowerCase()), topicName, topicName)
-      );
+      throw new KsqlException(
+          AclErrorMessageUtil.errorMessage(operation, topicName));
```

Comment:
I'm working on a similar patch that performs permissions checks for both the user and the KSQL server. In that patch I am thinking of changing this message to just throw a simple message like this:
```
      throw new KsqlException(String.format("Not authorized to %s topic(s): [%s]",
         StringUtils.capitalize(operation.toString().toLowerCase()), topicName)
      );
```

The idea behind it is to display this error in case KSQL cannot access the topic:
```
ksql> create stream s1(id int) with (kafka_topic='t_aug13', value_format='json');
The KSQL server cannot create stream: [s1]
Caused by: Not authorized to Read topic(s): [t_aug13]
```

What do you think about changing the message in `AclErrorMessageUtil` to the one proposed?

Reply:
How would the first error be generated for various different commands that run into ACLs? `The KSQL server cannot create stream: [s1]`

I do think the second line specific to ACLs is much more succinct than what I have currently and I'm fine with changing the Util code to
```
public static String errorMessage(
      final AclOperation operation, final Collection<String> topicNames) {
    final String topicNamesString = StringUtils.join(topicNames, ", ");
    return String.format(
        "Caused by: Not authorized to %s topic(s): [%s]",
        StringUtils.capitalize(
            operation.toString().toLowerCase()),
            topicNamesString
    );
  }
```

Reply:
The patch I'm writing will add the extra line to show `The KSQL server cannot create stream` in a different place. Anyway, I've been thinking more about this, and I found another idea to solve my problem.

If instead of using the `AclErrorMessageUtil` static method, what about creating our own `KsqlAuthorizationException` and throw that exception instead? The constructor could be something like:
```
public KsqlAuthorizationException(operation, topicException) {
	this.message = String.format("Authorization denied to %s on topic(s): %s", 
		StringUtils.capitalize(operation.toLowerCase()), topicException.unauthorizedTopics());
}

@Override
public String getMessage() {
   return this.message;
}
```

The above will display this:
```
ksql> create stream t_aug13(id int) with (kafka_topic='t_aug13', value_format='json');
Authorization denied to Read on topic(s): [t_aug13]
```
and when I append the extra message for the KSQL server, then I will get:
```
ksql> create stream t_aug13(id int) with (kafka_topic='t_aug13', value_format='json');
The KSQL (User:ksql) server cannot continue with the command execution.
Caused by: Authorization denied to Read on topic(s): [t_aug13]
```
`Caused by` is automatically added by the Exception if I print the complete message stack, so we can take advantage of it.

Btw, I'm planning to re-use the `KsqlAuthorizationException` to display authorization errors from SchemaRegistry as well, so it is going to save me code.